### PR TITLE
Use a global BLE lock.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/resources/ble_esp32.cc [hunk 1]
```diff
@@ -1382,19 +1398,17 @@ PRIMITIVE(create_peripheral_manager) {
...
+    ble_gatts_reset();
+    BleEventSource::instance()->on_event(peripheral_manager, kBleStarted);
   }
 
   return proxy;
 }
 
 PRIMITIVE(close) {
   ARGS(BleResourceGroup, group)
+  // The group is the only primitive that doesn't allocate a locker.
```
Comment (kasperl) [inline src/resources/ble_esp32.cc:1411]:
```suggestion
  // The close primitive is the only primitive that doesn't allocate a locker.
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/resources/ble_esp32.cc [hunk 2]
```diff
@@ -219,16 +248,10 @@ class BleReadWriteElement : public BleErrorCapableResource {
   }
 
   os_mbuf* mbuf_to_send() {
```
[the remark this answers is not in the document]
Comment (floitsch) [reply src/resources/ble_esp32.cc:250]: I don't think it's necessary. There was already the `mbuf_to_send` that just took the resource-group mutex.
And it's just important that there is a lock.

File: src/resources/ble_esp32.cc (hunk 1 above)
Comment (floitsch) [reply src/resources/ble_esp32.cc]: done.

**Pull request merged**